### PR TITLE
Prepare for gemini 3c

### PR DESCRIPTION
**Pull request by i1i1**
This pr updates monorepo and lands gemini 3c support. 3 commits which start with `WIP` are essentially one big commit which was split to several ones so that it will be easire to review (while only the last one compiles).

EDIT: This pr became too big. It would be best to see `git diff ec490c1 a257bf9` and review after `a257bf9`.

### CODE REVIEW

**Comment by ozgunozerk** [comment]
This is indeed a big PR, seems fine to me. 
One comment only: in the 1st and 3rd commit, you added A LOT of new implementations and definitions. I can't suggest a better way myself, but at first glance, it seemed like those stuff should be importable from the monorepo, so that you wouldn't have to write that much code in the SDK. 
I might be wrong, that's just an intuition

**Comment by ozgunozerk** [comment]
Since tests are failing and code is subject to change, I'll approve once the tests are fixed

**Comment by i1i1** [comment]
> in the 1st and 3rd commit, you added A LOT of new implementations and definitions. I can't suggest a better way myself, but at first glance, it seemed like those stuff should have been importable from the monorepo, so that you wouldn't have to write that much code in the SDK.

1st commit literally copies out unexported things from monorepo, I just copied out them for now so that we won't be blocked, but most of the things surely can be shared. As for the 3rd

> Since tests are failing and code is subject to change, I'll approve once the tests are fixed

Weird, they didn't fail locally. Restarted for now.

**Comment by ozgunozerk** [comment]
I think it was not "bad luck" the last time, still failing :/

File: src/node/mod.rs [hunk 1]
```diff
@@ -785,6 +784,69 @@ impl From<RpcMethods> for sc_service::RpcMethods {
...
+            bootstrap_nodes.clone(),
+        )
+        .boxed(),
+        request_response_protocols: vec![PieceByHashRequestHandler::create(move |req| {
+            get_piece_by_hash(req, &piece_cache)
+        })],
+        provider_storage,
+        reserved_peers: reserved_nodes
+            .into_iter()
+            .map(|a| a.to_string().parse().unwrap())
```
Comment (ozgunozerk) [inline src/node/mod.rs:842]: should this be expect instead?

**Comment by i1i1** [comment]
I think we can merge it as it is (Also I will squash it, as this pr is too big and messy).

**Review by ozgunozerk — APPROVED** [review]

**Comment by ozgunozerk** [comment]
Will steal unwrap lint for clippy 🏴‍☠️

**Comment by i1i1** [comment]
Test timed out with the following in logs:
```
2023-01-27T19:14:42.3813541Z 2023-01-27T19:14:34.580599Z  INFO sc_basic_authorship::basic_authorship: 🙌 Starting consensus session on top of parent 0x733c54fa18130bf7640f58eb9bd3d0dce48a3ca696eaa259f7b4bfbf314b05b6    
2023-01-27T19:14:42.3814402Z 2023-01-27T19:14:34.580705Z  INFO subspace: ⌛️ Discarding proposal for slot 1674846873; block production took too long    
2023-01-27T19:14:42.3815109Z 2023-01-27T19:14:34.580709Z  INFO subspace: 👉 Recompile your node in `--release` mode to mitigate this problem.    
2023-01-27T19:14:42.3816246Z 2023-01-27T19:14:34.586943Z  INFO sc_basic_authorship::basic_authorship: 🎁 Prepared block for proposing at 1 (4 ms) [hash: 0xc8dc352c77aa6604536a7dd96a5f7b11aed0cd5cdb7aa07cd8e7b8f7e3249e10; parent_hash: 0x733c…05b6; extrinsics (2): [0xb9fd…603e, 0x1632…ab71]]    
```

So ignoring test for now, as the github runners are too slow.

**Pull request merged**